packages/javascript-sdk/src/http-client/helpers.ts

```diff
@@ -182,6 +202,19 @@ export function normalizeIGJSON(res: Response): AuthorizationJSON {
   };
 }
 
+export function normalizeIGJSONResponseToAdviceJSON(res: Response): AuthorizationJSON {
```

Comment:
I think we should probably just accept the header needed in this function. We don't need to leak the entire response into this function since this only operates on headers.

I think we can therefore create a type that checks specifically for the WWW-Authenticate header.

```ts
type AuthenticateHeader = "WWW-Authenticate"
```

Reply:
this may not be as possible as i thought since repsonse headers are just strings and can't be narrowed from what i can tell

Reply:
however we can still narrow the scope of the parameter.

Reply:
Narrowing the scope of the parameter makes sense - I'll push this in a bit. Thanks for the suggestion.